app/scripts/modules/kubernetes/src/pipeline/stages/runJob/runJobStage.js

```diff
@@ -62,10 +49,36 @@ module.exports = angular.module('spinnaker.kubernetes.pipeline.stage.runJobStage
       this.stage.dnsPolicy = 'ClusterFirst';
     }
 
-    if (this.stage.container.imageDescription.fromTrigger === true) {
-      this.container = buildImageDescriptor(this.stage.container.imageDescription);
+    if (!this.stage.credentials && $scope.application.defaultCredentials.kubernetes) {
+      this.stage.credentials = $scope.application.defaultCredentials.kubernetes;
     }
 
+    this.contextImages = pipelineConfigService.getAllUpstreamDependencies(this.pipeline, this.stage).map((stage) => {
+      
+      if (stage.type !== 'findImage' && stage.type !== 'bake') {
+        return;
+      }
+
+      if (stage.type === 'findImage') {
+        return {
+          fromContext: true,
+          fromFindImage: true,
+          cluster: stage.cluster,
+          pattern: stage.imageNamePattern,
+          repository: stage.name,
+          stageId: stage.refId
+        };
+      }
+
+      return {
+        fromContext: true,
+        fromBake: true,
+        repository: stage.ami_name,
+        organization: stage.organization,
+        stageId: stage.refId
+      };
+    });
```

Comment:
I think you need to filter out `undefined` from this list (in the case where you had an upstream stage whose type wasn't `findImage` or `bake`). Just tag a `.filter(image => !!image)` to the end.

Reply:
👍